tests/dist.rs

```diff
@@ -1984,3 +1984,68 @@ fn checks_files_hashes_before_reuse() {
         assert!(noticed_bad_checksum.get());
     })
 }
+
+#[test]
+fn handle_corrupt_partial_downloads() {
+    setup(None, false, &|url,
+                         toolchain,
+                         prefix,
+                         download_cfg,
+                         temp_cfg| {
+        // write a corrupt partial out
+        let path = url.to_file_path().unwrap();
+        let target_hash = utils::read_file(
+            "target hash",
+            &path.join("dist/2016-02-02/rustc-nightly-x86_64-apple-darwin.tar.gz.sha256"),
+        )
+        .unwrap()[..64]
```

Comment:
This is unusual, I think I'd prefer:

```suggestion
        .unwrap().trim()
```

Assuming I've correctly understood the purpose of that `[..64]`

Reply:
Turns out the hash file has more just the sha in it:
`66d22b7dea5bf8af3295ef13a2ac71abe85e9ad2616eeb22623ca85e6f613ccf *rustc-nightly-x86_64-apple-darwin.tar.gz`

We could split the string on whitespace but at the same time we know it is a fixed 64 character string so I'm tempted to leave it as is?

Also, in the actual code where it reads a hash file (src/dist/download.rs - fn download_hash) it just uses a slice too.

Reply:
Raised MR to glamhoth (https://github.com/Glamhoth/rustup.rs/pull/3) to use a contstant for the slice.